fiftyone/core/collections.py

```diff
@@ -9155,6 +9156,39 @@ def values(
         Returns:
             the list of values
         """
+
+        # If we do not need to follow insertion order, we can potentially use a
+        # covered index query to get the values directly from the index and
+        # avoid a COLLSCAN.
+        if not _enforce_natural_order:
+            field = None
+            if isinstance(field_or_expr, str):
+                field = field_or_expr
+            elif isinstance(field_or_expr, list) and len(field_or_expr) == 1:
+                field = field_or_expr[0]
+
+            # Checking _is_full_collection last as it could potentially be expensive.
+            # In the future, we might be less restrictive, but that needs
+            # further investigation.
+            if (
+                field in ["id", "_id", "filepath"]
+                and self._is_full_collection()
+            ):
+                try:
+                    return foo.get_indexed_values(
+                        self._dataset._sample_collection,
+                        field,
+                        values_only=True,
+                    )
+                except ValueError as e:
+                    # Fallback to values, but log the error to surface the
+                    # recommendation to create an index
+                    logger.debug(e)
+                except Exception as e:
+                    # Unexpected error, so we should surface it in the logs
```

Comment:
I disagree with this. Why should unexpected errors be tolerated in `get_indexed_values()` specifically? I think they should be raised like they are everywhere else.

Reply:
Unless I am missing something about `get_indexed_values()` where it might not work deterministically?

Reply:
`get_indexed_values()` itself will raise all errors, but I think it should be tolerated inside values() and only raise if values() also errors

Reply:
What's an example of a non-`ValueError` that `get_indexed_values()` could raise where it makes sense to gracefully continue here? 

My claim is that anything other than the specific "you should add an index" error that we're logging as a debug message represents a logical error in the implementation of `get_indexed_values()` that needs to be fixed, not ignored.

Reply:
ok I'm fine with raising unanticipated errors since they must specifically add  `_enforce_natural_order=False`, so anyone using normal values() cannot be impacted